app/src/main/java/com/yacgroup/yacguide/BaseNavigationActivity.kt

```diff
@@ -62,7 +51,15 @@ abstract class BaseNavigationActivity : AppCompatActivity(), NavigationView.OnNa
 
         navView.setNavigationItemSelectedListener(this)
 
-        onBackPressedDispatcher.addCallback(this, onBackPressedCallback)
+        onBackPressedDispatcher.addCallback(this, object: OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() {
+                if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
```

Comment:
@paetz: Here I'm bit unsure if this is a good programming style.  `drawerLayout` is captured on a closure.

Reply:
I have no clue, either.

Reply:
I'm also a bit surprised that this works anyway because you declare a callback function which is unrelated to the `onCreate` context. However, if you would use the `val drawerLayout = ...` you get a warning that the variable is shadowed by the outer scope. Nevertheless, it works :-).